client/app/bundles/course/survey/containers/QuestionFormDialogue/QuestionForm.jsx

```diff
@@ -89,254 +98,387 @@ const questionFormTranslations = defineMessages({
-    if (values.min_options >= filledOptions) {
-      errors.min_options = questionFormTranslations.lessThanFilledOptions;
-    } else if (values.min_options < 0) {
-      errors.min_options = questionFormTranslations.atLeastZero;
-    }
-  }
+  const questionType = watch('question_type');
+  const options = watch('options');
+  const deletedOptions = watch('optionsToDelete');
 
-  if (values.question_type === MULTIPLE_RESPONSE && values.max_options) {
-    if (values.max_options < 1) {
-      errors.max_options = questionFormTranslations.atLeastOne;
-    } else if (values.max_options > filledOptions) {
-      errors.max_options = questionFormTranslations.noMoreThanFilledOptions;
-    } else if (values.min_options && values.min_options > values.max_options) {
-      errors.max_options = questionFormTranslations.notLessThanMin;
-    }
-  }
+  // When the values in any of the array options fields are changed,
+  // 'fields' from useFieldArray are not updated but the internal values of options
+  // are already updated in useForm. We then use watch to extract the updated options values
+  // and update those to controlledFields as seen below.
+  const controlledOptionsFields = optionsFields.map((field, index) => ({
+    ...field,
+    ...options[index],
+  }));
 
-  return errors;
-};
+  const controlledDeletedOptionsFields = deletedOptionsFields.map(
+    (field, index) => ({
+      ...field,
+      ...deletedOptions[index],
+    }),
+  );
 
-class QuestionForm extends Component {
-  renderMultipleChoiceFields() {
-    return (
-      <div>
-        {this.renderTiledViewToggle()}
-        {this.renderValidOptionCount()}
-        {this.renderOptionFields({ multipleChoice: true })}
-      </div>
-    );
-  }
-
-  renderMultipleResponseFields() {
-    const { intl } = this.props;
-    return (
-      <div>
-        {this.renderTiledViewToggle()}
-        <div style={styles.numberOfResponsesDiv}>
-          {this.renderValidOptionCount()}
-          {this.renderNumberOfResponsesField(
-            'min_options',
-            intl.formatMessage(translations.minOptions),
-          )}
-          {this.renderNumberOfResponsesField(
-            'max_options',
-            intl.formatMessage(translations.maxOptions),
-          )}
-        </div>
-        {this.renderOptionFields({ multipleResponse: true })}
-      </div>
-    );
-  }
-
-  renderNumberOfResponsesField(name, label) {
-    const { disabled } = this.props;
-    return (
-      <Field
-        component={renderTextField}
-        type="number"
-        {...styles.numberOfResponsesField}
-        placeholder={questionFormTranslations.noRestriction.defaultMessage}
-        {...{ name, label, disabled }}
-      />
-    );
-  }
+  useEffect(() => {
+    if (optionsFields.length === 0) {
+      optionsAppend({
+        weight: null,
+        option: '',
+        image_url: '',
+        image_name: '',
+        file: null,
+      });
+    }
+  }, [optionsFields.length === 0]);
```

Comment:
Hmm this is interesting haha.

Reply:
This is done so that when all options are deleted, a new option is created so that it has at least 1 option for the user to input.